vkubelet/pod.go

```diff
@@ -29,138 +25,18 @@ func addPodAttributes(span *trace.Span, pod *corev1.Pod) {
-	ctx, span := trace.StartSpan(ctx, "onDeletePod")
-	defer span.End()
-	logger := log.G(ctx).WithField("method", "onDeletePod")
-
-	pod, ok := obj.(*corev1.Pod)
-	if !ok {
-		delta, ok := obj.(cache.DeletedFinalStateUnknown)
-		if !ok {
-			span.SetStatus(trace.Status{Code: trace.StatusCodeInvalidArgument, Message: fmt.Sprintf("Unexpected object from event: %T", obj)})
-			logger.Errorf("obj is not of a valid type: %T", obj)
-			return
-		}
-
-		if pod, ok = delta.Obj.(*corev1.Pod); !ok {
-			span.SetStatus(trace.Status{Code: trace.StatusCodeInvalidArgument, Message: fmt.Sprintf("Unexpected object from event: %T", obj)})
-			logger.Errorf("obj is not of a valid type: %T", obj)
-			return
-		}
-	}
-
-	addPodAttributes(span, pod)
-
-	logger.Debugf("Receive deleted pod '%s/%s'", pod.GetNamespace(), pod.GetName())
-
-	if s.resourceManager.DeletePod(pod) {
-		span.Annotate(nil, "Add pod to synchronizer channel.")
-		select {
-		case <-ctx.Done():
-			logger = logger.WithField("pod", pod.GetName()).WithField("namespace", pod.GetNamespace())
-			logger.WithError(ctx.Err()).Debug("Cancel send pod event due to cancelled context")
-			return
-		case s.podCh <- &podNotification{pod: pod, ctx: ctx}:
-		}
-	}
-}
-
-func (s *Server) startPodSynchronizer(ctx context.Context, id int) {
-	logger := log.G(ctx).WithField("method", "startPodSynchronizer").WithField("podSynchronizer", id)
-	logger.Debug("Start pod synchronizer")
-
-	for {
-		select {
-		case <-ctx.Done():
-			logger.Info("Stop pod syncronizer")
-			return
-		case event := <-s.podCh:
-			s.syncPod(event.ctx, event.pod)
-		}
-	}
-}
-
-func (s *Server) syncPod(ctx context.Context, pod *corev1.Pod) {
-	ctx, span := trace.StartSpan(ctx, "syncPod")
-	defer span.End()
-	logger := log.G(ctx).WithField("pod", pod.GetName()).WithField("namespace", pod.GetNamespace())
-
-	addPodAttributes(span, pod)
-
-	if pod.DeletionTimestamp != nil {
-		span.Annotate(nil, "Delete pod")
-		logger.Debugf("Deleting pod")
-		if err := s.deletePod(ctx, pod); err != nil {
-			logger.WithError(err).Error("Failed to delete pod")
-		}
-	} else {
-		span.Annotate(nil, "Create pod")
-		logger.Debugf("Creating pod")
-		if err := s.createPod(ctx, pod); err != nil {
-			logger.WithError(err).Errorf("Failed to create pod")
-		}
+func (s *Server) createOrUpdatePod(ctx context.Context, pod *corev1.Pod) error {
+	// Check if the pod is already known by the provider.
+	// NOTE: Some providers return a non-nil error in their GetPod implementation when the pod is not found while some other don't.
+	// Hence, we ignore the error and just act upon the pod if it is non-nil (meaning that the provider still knows about the pod).
+	if pp, _ := s.provider.GetPod(ctx, pod.Namespace, pod.Name); pp != nil {
+		// The pod has already been created in the provider.
+		// Hence, we return since pod updates are not yet supported.
+		log.G(ctx).Warnf("skipping update of pod %s as pod updates are not supported", pp.Name)
```

Comment:
Nit, should use `log.G(ctx).WithField("pod", <podName>)` (could probably even set this in the context before passing it through).

Reply:
Taking notes here for working on this during #348.